hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdb/util/ManagedSstFileReader.java

```diff
@@ -46,80 +52,176 @@ public class ManagedSstFileReader {
   public ManagedSstFileReader(final Collection<String> sstFiles) {
     this.sstFiles = sstFiles;
   }
-  public Stream<String> getKeyStream() throws RocksDBException {
-    final ManagedSstFileIterator itr = new ManagedSstFileIterator(sstFiles);
-    final Spliterator<String> spliterator = Spliterators
-        .spliteratorUnknownSize(itr, 0);
-    return StreamSupport.stream(spliterator, false).onClose(itr::close);
+
+  public static <T> Stream<T> getStreamFromIterator(
+          CloseableIterator<T> itr) {
+    final Spliterator<T> spliterator = Spliterators
+            .spliteratorUnknownSize(itr, 0);
+    return StreamSupport.stream(spliterator, false).onClose(() -> {
+      try {
+        itr.close();
+      } catch (IOException e) {
+        throw new UncheckedIOException(e);
+      }
+    });
+  }
+
+  public Stream<String> getKeyStream() throws RocksDBException,
```

Comment:
nit: line break in your code is indented by 8 spaces instead of 4. Please configure your IDE accordingly and indent the code that way.

Reply:
@swamirishi Pls import `./hadoop-ozone/dev-support/intellij/ozone-style.xml` and use it in your IntelliJ IDE.

https://www.jetbrains.com/help/idea/configuring-code-style.html#import-code-style

Reply:
done